zerver/tests/test_signup.py

```diff
@@ -1587,54 +1584,165 @@ def test_registration_through_ldap(self):
+        self.assert_in_response("Check your email so we can get started.", result)
+
+        with self.settings(
+                POPULATE_PROFILE_VIA_LDAP=True,
+                LDAP_APPEND_DOMAIN='zulip.com',
+                AUTH_LDAP_BIND_PASSWORD='',
+                AUTH_LDAP_USER_ATTR_MAP=ldap_user_attr_map,
+                AUTHENTICATION_BACKENDS=('zproject.backends.ZulipLDAPAuthBackend',),
+                AUTH_LDAP_USER_DN_TEMPLATE='uid=%(user)s,ou=users,dc=zulip,dc=com'):
+
+            # Click confirmation link. This will 'authenticated_full_name'
+            # session variable which will be used to set the fullname of
+            # the user.
+            result = self.submit_reg_form_for_user(email,
+                                                   password,
+                                                   full_name="Ignore",
+                                                   from_confirmation="1",
+                                                   # Pass HTTP_HOST for the target subdomain
+                                                   HTTP_HOST=subdomain + ".testserver")
+
+        with patch('zerver.views.registration.name_changes_disabled', return_value=True):
+            result = self.submit_reg_form_for_user(email,
+                                                   password,
+                                                   # Pass HTTP_HOST for the target subdomain
+                                                   HTTP_HOST=subdomain + ".testserver")
+            user_profile = UserProfile.objects.get(email=email)
+            # Name comes form session.
+            self.assertEqual(user_profile.full_name, 'New LDAP fullname')
+
+    def test_registration_when_names_changes_are_disabled(self):
+        # type: () -> None
+        password = "testing"
+        email = "newuser@zulip.com"
+        subdomain = "zulip"
+
+        with patch('zerver.views.registration.get_subdomain', return_value=subdomain):
+            result = self.client_post('/register/', {'email': email})
+
+        self.assertEqual(result.status_code, 302)
+        self.assertTrue(result["Location"].endswith(
+            "/accounts/send_confirm/%s" % (email,)))
+        result = self.client_get(result["Location"])
+        self.assert_in_response("Check your email so we can get started.", result)
+
+        with patch('zerver.views.registration.name_changes_disabled', return_value=True):
+            result = self.submit_reg_form_for_user(email,
+                                                   password,
+                                                   full_name="New Name",
+                                                   # Pass HTTP_HOST for the target subdomain
+                                                   HTTP_HOST=subdomain + ".testserver")
+            user_profile = UserProfile.objects.get(email=email)
+            # Name comes form session.
+            self.assertEqual(user_profile.full_name, 'New Name')
```

Comment:
I think the intent of the code is that if `name_changes_disabled` is True, we should be getting the full name from LDAP and ignoring the argument if passed in by the user.  So either this testing is wrong or the code is wrong...

Reply:
I think this test tests `name_changes_disabled`  when we are not under LDAP. Let me add comments for clarification.